micropub.php

```diff
@@ -882,13 +882,36 @@ public static function default_file_handler( $post_id ) {
 	 * https://ownyourswarm.p3k.io/docs#checkins
 	 *
 	 * WordPress geo data is stored in post meta: geo_address (free text),
-	 * geo_latitude, and geo_longitude:
+	 * geo_latitude, geo_longitude, and geo_public:
 	 * https://codex.wordpress.org/Geodata
+	 * It is noted that should the HTML5 style geolocation properties of altitude, accuracy, speed, and heading are
+	 * used they would use the same geo prefix. Simple Location stores these when available using accuracy to estimate
+	 * map zoom when displayed.
 	 */
 	public static function store_geodata( $args ) {
 		$properties = static::get( static::$input, 'properties' );
 		$location   = static::get( $properties, 'location', static::get( $properties, 'checkin' ) );
 		$location   = static::get( $location, 0, null );
+		// Location-visibility is an experimental property https://indieweb.org/Micropub-extensions#Location_Visibility
+		// It attempts to mimic the geo_public
+		$visibility = static::get( $properties, 'location-visibility', null );
+		if ( $visibility ) {
+			$visibility = array_pop( $visibility );
+			if ( ! isset( $args['meta_input'] ) ) {
+				$args['meta_input'] = array();
+			}
+			switch ( $visibility ) {
+				case 'public':
+					$args['meta_input']['geo_public'] = 1;
+					break;
+				case 'private':
+					$args['meta_input']['geo_public'] = 0;
+					break;
+				case 'protected':
+					$args['meta_input']['geo_public'] = 2;
+					break;
+			}
```

Comment:
we should probably return an error if it's something other than these three values, right?

Reply:
i meant the `location-visibility` micropub property values. https://indieweb.org/Micropub-extensions#Location_Visibility only lists `public`, `protected`, and `private`, so we should probably return 400 if we get anything else.